server-common/src/main/java/org/apache/kafka/server/common/TransactionVersion.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.server.common;
+
+import java.util.Collections;
+import java.util.Map;
+
+public enum TransactionVersion implements FeatureVersion {
+
+    // Version 1 enables flexible transactional state records. (KIP-890)
+    TV_1(1, MetadataVersion.IBP_4_0_IV1, Collections.emptyMap()),
```

Comment:
I wonder why we need a version to enable flexible records. Couldn't we use the version supporting it when we enable version 2? I may have missed something.

Reply:
This was a discussion I had with @cmccabe. I think the idea was that even if we want to disable transaction version, we may still want to keep our flexible records due to the compatibility reasons. I know Jeff worked on that KIP to ignore the fields, but this was an extra safeguard.

Reply:
Understood. As you said, this is not strictly necessary but it does not hurt.